src/graphics/colour.rs

```diff
@@ -0,0 +1,67 @@
+type ColourRGB = (u8, u8, u8);
+
+/// Enum representing the colour options available for gates
+#[derive(Debug, Copy)]
+pub enum Colour {
+    /// The primary colour of the gates (in a light theme, this would be black)
+    Foreground,
+
+    /// The background colour (in a light theme, this would be white)
+    Background,
+
+    /// Error colour (e.g. a FSM not enabled)
+    Alert,
+
+    /// Info colour (e.g. counter output in denary)
+    Info,
+
+    /// Arbitrary RGB colour
+    RGB(u8, u8, u8)
```

Comment:
This should probably be `ColorRGB`:

```suggestion
    RGB(ColourRGB)
```